### ISSUE TITLE
Allow changing FROM image on command line

### ISSUE THREAD

**Issue by tnozicka** [containers/buildah#2007]
**Description**

It is often the case that I, and other people in the OpenShift org, need to change the image that is used in the Dockerfile and use a different one. OpenShift builds are based on that idea and given you have input imagestreams it preprocesses the Dockerfile and replaces the FROM line. The way how we would use such dockerfiles locally is to use https://github.com/openshift/imagebuilder to build it like:
```
imagebuilder --from registry.svc.ci.openshift.org/ocp/ubi:7 -mount ~/projects/origin-repos/4.3/ocp-4.3-default.repo:/etc/yum.repos.d/ocp-4.3-default.repo -mount ~/projects/origin-repos/4.3/ops-mirror.pem:/etc/yum.repos.d/ops-mirror.pem -t openshift-base -f base/Dockerfile base/
```
unfortunately using imagebuilder implies using Docker and being root (https://github.com/openshift/imagebuilder/issues/128). I really like using podman/buildah and being just a regular user, but this use cases forces me back into Docker and being root.

**Describe the results you expected:**
support `--from` flag in buildah and `podman build` to be able to replace the FROM image in Dockerfile

cc @dustymabe @vrothberg

**Comment by vrothberg** [comment]
Thanks for opening the issue, @tnozicka! I think this fairly easy to support and a quite common use-case in the OpenShift land.

@TomSweeneyRedHat @nalind @rhatdan WDYT?

**Comment by TomSweeneyRedHat** [comment]
@tnozicka just to make sure I'm on the same page, given a Dockerfile like:
```
FROM alpine
ARG FIRSTARG
```

Then a command like "buildah bud --from busybox" would in effect make the processed Dockerfile look like:
```
FROM busybox
ARG FIRSTARG
```

But the actual Dockerfile file on disk would not change?  If that's the case, I think it's doable.

**Comment by tnozicka** [comment]
> But the actual Dockerfile file on disk would not change?

correct, no disk changes

thanks for looking into it!

**Comment by tnozicka** [comment]
```
FROM alpine AS alpine-stage
```
Aliases for multi stage build like `alpine-stage` would still retain its name even when running `buildah bud --from busybox` , right?

**Comment by TomSweeneyRedHat** [comment]
@tnozicka I'm glad you brought that up, I probably would have just wiped the entire thing.  But we certainly could retain the "AS alpine-stage" portion of the FROM statement if it was there.  Just a little bit more processing, nothing major.

**Comment by denysvitali** [comment]
Related: https://github.com/containers/buildah/issues/1919

**Comment by dustymabe** [comment]
bump - I hit a case where I could have used this feature today :) - thanks for all the great work you all do!

**Comment by rhatdan** [comment]
@ashley-cui PTAL

**Comment by TomSweeneyRedHat** [comment]
@rhatdan Let me take another look in here.  The sample Dockerfile case up top is cake, the problem that we ran into and discussed within the PR is the value replaced for the FROM should only be passed into the last FROM in the Dockerfile when there are multiple FROM statements in the Dockerfile.  The issue is related to counting the number of steps issue that we have, it was hard to pre-determine how many steps and which FROM was the last FROM.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Did you ever look at this?

**Comment by TomSweeneyRedHat** [comment]
Not yet, still on the list.

**Comment by TomSweeneyRedHat** [comment]
Addressed with  #2823 closing
